Src/Base/AMReX_GpuAtomic.H

```diff
@@ -22,13 +22,13 @@ namespace detail {
 
     template <typename R, typename I, typename F>
     AMREX_GPU_DEVICE AMREX_FORCE_INLINE
-    R atomic_op (R* address, R val, F f) noexcept
+    R atomic_op (R* const address, R const val, F const f) noexcept
```

Comment:
From
```C++
#include <type_traits>
```
one should probably rather take:
```suggestion
    R atomic_op (std::add_const<R*>::type address, std::add_const<R>::type val, std::add_const<F>::type f) noexcept
```